specs/validator/beacon_node_oapi.yaml

```diff
@@ -0,0 +1,609 @@
+      - name: slot
+        in: query
+        required: true
+        description: "The slot for which the block should be proposed."
+        schema:
+          type: integer
+          format: uint64
+      - name: randao_reveal
+        in: query
+        required: true
+        description: "The validator's randao reveal value."
+        schema:
+          type: string
+          format: byte
+      responses:
+        200:
+          description: Success response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/BeaconBlock'
+        400:
+          $ref: '#/components/responses/InvalidRequest'
+        500:
+          $ref: '#/components/responses/InternalError'
+        503:
+          $ref: '#/components/responses/CurrentlySyncing'
+    post:
+      tags:
+        - MinimalSet
+      summary: "Publish a signed block."
+      description: "Instructs the beacon node to publish a newly signed beacon block to the beacon network, to be included in the beacon chain."
+      parameters:
+      - name: beacon_block
+        in: query
+        required: true
+        description: "The `BeaconBlock` object, as sent from the beacon node originally, but now with the signature field completed."
+        schema:
+          $ref: '#/components/schemas/BeaconBlock'
+      responses:
+        200:
+          $ref: '#/components/responses/Success'
+        400:
+          $ref: '#/components/responses/InvalidRequest'
+        500:
+          $ref: '#/components/responses/InternalError'
+        503:
+          $ref: '#/components/responses/CurrentlySyncing'
+
+  /validator/attestation:
+    get:
+      tags:
+        - MinimalSet
+      summary: "Produce an attestation, without signature."
+      description: "Requests that the beacon node produce an IndexedAttestation, with a blank signature field, which the validator will then sign."
+      parameters:
+      - name: slot
+        in: query
+        required: true
+        description: "The slot for which the attestation should be proposed."
+        schema:
+          type: integer
+      - name: shard
+        in: query
+        required: true
+        description: "The shard number for which the attestation is to be proposed."
+        schema:
+          type: integer
+      responses:
+        200:
+          description: Success response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IndexedAttestation'
+        400:
+          $ref: '#/components/responses/InvalidRequest'
+        500:
+          $ref: '#/components/responses/InternalError'
+        503:
+          $ref: '#/components/responses/CurrentlySyncing'
+    post:
+      tags:
+        - MinimalSet
+      summary: "Published a signed attestation."
+      description: "Instructs the beacon node to publish a newly signed IndexedAttestation object, to be incorporated into the beacon chain."
```

Comment:
This is to be broadcast on the particular shard subnet. Let's that mention that here.

As for "to be incorporated into the beacon chain", it is only incorporated into the local chain when it is seen on a subsequent block rather than immediately added to local chain. Might just remove this part of the statement.